code/datums/components/squeak.dm

```diff
@@ -41,6 +41,10 @@
 		use_delay = use_delay_override
 
 /datum/component/squeak/proc/play_squeak()
+	if(ismob(parent))
```

Comment:
Might want to just unregister on death. And register back when the mouse comes back alive. Unsure how viable that is though

Reply:
I did explore during that, but then I realized there are a few different methods it could come alive again, from staves of healing, admin rejuv, lazarus, etc.
I felt that trying to remove the component and then re-add it in each possible case would just lead to more bugs and edge cases. For example, worries about incorrectly adding the component multiple times.
For all of these reasons I felt it was cleaner, clearer and less bug-prone to just disable the squeaking while the mob is dead.

Reply:
we kinda need one way to revive things. Or atleast have the ways to revive them call a signal or such.

Reply:
`update_revive` is the standard way to revive a mob